localstack-core/localstack/services/events/target.py

```diff
@@ -102,14 +103,18 @@ def __init__(
         rule_arn: Arn,
         rule_name: RuleName,
         service: str,
+        region: str,
+        account_id: str,
     ):
         self.target = target
         self.rule_arn = rule_arn
         self.rule_name = rule_name
         self.service = service
+        self.region = region
+        self.account_id = account_id
 
-        self.region = extract_region_from_arn(self.target["Arn"])
-        self.account_id = extract_account_id_from_arn(self.target["Arn"])
+        self.target_region = extract_region_from_arn(self.target["Arn"])
```

Comment:
Consider using a `@cached_property` for `target_*` https://docs.python.org/3/library/functools.html#functools.cached_property
Makes these properties immutable and seems more intention-revealing for a dependent value rather than a separate field